### PR TITLE
Fix manifest workflow failure

### LINKED ISSUE #2854: [Bug]: Version workflow to auto update input manifests is failing
### Describe the bug

Our version workflow to auto raise PR for input manifest update is currently failing for almost 10 days. https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml
The context of error:
```
BUILD FAILED in 11s
2022-11-04 00:49:45 WARNING  Error getting version of performance-analyzer: Command './gradlew properties -Dopensearch.version=3.0.0 -Dbuild.snapshot=false' returned non-zero exit status 1., ignored
2022-11-04 00:49:45 INFO     Checking out opensearch-observability#main
2022-11-04 00:49:45 INFO     Executing "git init" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git remote add origin https://github.com/opensearch-project/opensearch-observability.git" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git fetch --depth 1 origin main" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Removing /tmp/tmpco2c3n05
```
The problem might be the workflow trying to clone `https://github.com/opensearch-project/opensearch-observability.git` but the correct URL should be `https://github.com/opensearch-project/observability.git`
### To reproduce

https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml

### Expected behavior

_No response_

### Host / Environment

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by rishabh6788**
Signed-off-by: Rishabh Singh <sngri@amazon.com>

### Description
This PR fixes version workflow to auto update input manifests.

### Issues Resolved
#2854 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/manifests_workflow/input_manifests.py [hunk 1]
```diff
@@ -99,21 +100,23 @@ def update(
...
-                    )
-
-                    component_version = component.version
-                    if component_version:
-                        release_version = ".".join(component_version.split(".")[:3])
-                        if release_version not in main_versions.keys():
-                            main_versions[release_version] = []
-                        main_versions[release_version].append(component)
-                        logging.info(f"{component.name}#main is version {release_version} (from {component_version})")
+                    if type(component) is ComponentFromSource:
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply src/manifests_workflow/input_manifests.py:103]: yes, return repo for only those components that have `repository` attribute

File: src/manifests_workflow/component_opensearch.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ def checkout(
         name: str,
         path: str,
         opensearch_version: str,
+        repo_url: str,
```
Comment (zelinh) [inline src/manifests_workflow/component_opensearch.py:24]: Shall we also update the `GitRepository` below? This parameter seems not be used and picked up from the `GitRepository` initialization.

Comment (rishabh6788) [reply src/manifests_workflow/component_opensearch.py:24]: This has not been removed to preserve the old interface to make sure other tests are not affected by this change. @zelinh

Comment (zelinh) [reply src/manifests_workflow/component_opensearch.py:24]: Had an offline discussion with @rishabh6788 so there was a miss and updated in the latest commit below.

**Review by zelinh — APPROVED** [review]
I am good with this PR to unblock the manifest workflow. Any other opinions here or we would merge this PR. Thanks.
@dblock @gaiksaya @prudhvigodithi

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**